jina/types/arrays/neural_ops.py

```diff
@@ -82,3 +83,26 @@ def match(
                 d = Document(darray[int(_id)], copy=True)
                 d.scores[m_name] = _dist
                 _q.matches.append(d)
+
+    def visualize(
+        self, colors: Union[None, np.ndarray] = None, file_path: Union[None, str] = None
+    ):
+        """Visualize embeddings in a 2D projection with the PCA algorithm.
+
+        :param colors: Optional np.ndarray of lengh equal to len(self) containing integers for colors.
+        :param file_path: Optional path to store the visualization.
+        """
+
+        import matplotlib.pyplot as plt
+
+        pca = PCA(n_components=2)
```

Comment:
`n_components` should be the variable you initialized in constructor?

Reply:
which constructor? It is in the PCA constructor, not in method.visualize